### DISCUSSION TITLE
IIS Module: which files are tracked for triggering a restart?

### DISCUSSION THREAD

**Discussion by tbolon** [dotnet/aspnetcore#54191] [Q&A]
I have trouble understanding which files are monitored (and by who) for IIS to decide recycling a running ASP.NET Core app.

Previously (ASP era) web.config, bin/* files and app_offline.htm were monitored.

I have browsed the ANCM source code but I was not able to find any code responsible for watching file modifications, only code for the app_offline.htm scenario.

It seems IIS detect web.config changes by itself, governed by the [fcnMode](https://stackoverflow.com/a/31470150) on httpRuntime.

My question is: is the web.config file the only one tracked for apprestart?

Full story: my goal is to allow our developers to be able to work the same way they did pre-core era, where they simply have a IIS site pointing to their local .csproj directory, with a web.config ready to serve this app (as soon as it's compiled at least once) using shadow copy. As soon as they change and recompile something, IIS will detect a change then restart the app. I was able to create a correct web.config for IIS, but I can't find a way to trigger a restart of the website just by recompiling the program. The app will restart as soon as I touch the web.config file.

**Comment by lextm** [comment]
> My question is: is the web.config file the only one tracked for apprestart?

IIS only checks `<system.webServer>` changes. The link you referred to mostly talked about the old ASP.NET and `<system.web>` which is irrelevant to ASP.NET Core.

> Full story: my goal is to allow our developers to be able to work the same way they did pre-core era, where they simply have a IIS site pointing to their local .csproj directory, with a web.config ready to serve this app (as soon as it's compiled at least once) using shadow copy. As soon as they change and recompile something, IIS will detect a change then restart the app. I was able to create a correct web.config for IIS, but I can't find a way to trigger a restart of the website just by recompiling the program. The app will restart as soon as I touch the web.config file.

That's in vain. ASP.NET Core was designed to be different and developers need to adapt to the new world in which things like [VS IIS development-time support](https://learn.microsoft.com/aspnet/core/host-and-deploy/iis/development-time-iis-support) are essential parts.
